Mathlib/LinearAlgebra/Matrix/PosDef.lean

```diff
@@ -470,21 +470,50 @@ theorem toQuadraticForm' [DecidableEq n] {M : Matrix n n ℝ} (hM : M.PosDef) :
     toLinearMap₂'_apply']
   apply hM.2 x hx
 
-/-- The eigenvalues of a positive definite matrix are positive. -/
-lemma eigenvalues_pos [DecidableEq n] {A : Matrix n n 𝕜}
-    (hA : Matrix.PosDef A) (i : n) : 0 < hA.1.eigenvalues i := by
-  simp only [hA.1.eigenvalues_eq]
-  exact hA.re_dotProduct_pos <| (ofLp_eq_zero 2).ne.2 <|
-    hA.1.eigenvectorBasis.orthonormal.ne_zero i
+section conjugate
+variable [DecidableEq n] {x U : Matrix n n R}
 
+/-- For an invertible matrix `U`, `star U * x * U` is positive definite iff `x` is.
+This works on any ⋆-ring with a partial order.
+
+See `IsUnit.isStrictlyPositive_star_left_conjugate_iff'` for a similar statement for star-ordered
+rings. For matrices, positive definiteness is equivalent to strict positivity when the underlying
+field is `ℝ` or `ℂ` (see `Matrix.isStrictlyPositive_iff_posDef`). -/
+theorem _root_.Matrix.IsUnit.posDef_star_left_conjugate_iff (hU : IsUnit U) :
+    PosDef (star U * x * U) ↔ x.PosDef := by
```

Comment:
I realize you're simply moving this proof, but can't you simply use [Matrix.PosDef.conjTranspose_mul_mul_same](https://leanprover-community.github.io/mathlib4_docs/Mathlib/LinearAlgebra/Matrix/PosDef.html#Matrix.PosDef.conjTranspose_mul_mul_same) twice to prove this more easily? Probably lifting `U` to `(Matrix n n R)ˣ` will make the forward direction easier.